aws/protocol/query/array.go

```diff
@@ -0,0 +1,37 @@
+package query
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// Array represents the encoding of Query lists and sets
+type Array struct {
+	values url.Values
+	prefix string
+	flat bool
+	memberName string
+	size int32
+}
+
+func newArray(values url.Values, prefix string, flat bool, memberName string) *Array {
+	return &Array{
+		values: values,
+		prefix: prefix,
+		flat: flat,
+		memberName: memberName,
+	}
+}
+
+// Value adds a new element to the Query Array. Returns a Value type used to
+// encode the array element.
```

Comment:
Would be helpful to document an example for each `flattened` vs `unflattened` array  and it's encoded value. 

Can optionally provide a link to smithy docs query serialization for further reference.